### ISSUE TITLE
Basic variable/string manipulation (unit files)

### ISSUE THREAD

**Issue by bjne** [systemd/systemd#5649]
### Submission type
  - [x] Request for enhancement (RFE)

While working on systemd service files for openstack (api) services, I found that all these services
can be started in a very generic way, that is easily "templated", and hence would work great with
unit files. The problem I faced tho, was systemd/service/unit files does not seem to support any type
of string manipulation/filters, and, given that some of the design goals of systemd was to prevent forking
all kinds of (useless) scripts, I do not like having to fork a bash shell just for splitting a variable.

Some of the background here is that my services are split in two parts, project and module, say for
example keystone-api. The service is then openstack@keystone-api and %i is keystone-api

To be able to find the configuration files and the python venv location and so on, I need two variables
available in my unit file, project=keystone and module=api

currently the way I accomplish is by:

 ExecStartPre=/bin/bash -c 'unit="%i" ; \
  systemctl set-environment project=$${unit%%-*}  module=$${unit#*-}'

This is both impractical and ugly.

Any chance systemd could implement very basic string manipulation?
%{i|split(-)[0]} would be really nice.

**Comment by poettering** [comment]
Well, systemd unit files are explicitly not supposed to be a scripting or macro language. If you want that, then please use m4 or shell or something like that. After all, you can invoke anything you like from systemd unit files.

It's definitely our goal to avoid any further language compleixities with unit files, in fact we'd rather dumb it down, than add more tricks to it.

Sorry, but I think there's nothing wrong with using a shell or m4 if that's what you actually need.

Sorry, closing.
